addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1096,18 +1096,31 @@ const Wysiwyg = Widget.extend({
                             backgroundGradient = backgroundImage;
                         }
                     }
+                    const hadSelection = range && (range.startContainer !== range.endContainer || range.startOffset !== range.endOffset);
                     colorpicker = new ColorPaletteWidget(this, {
                         excluded: ['transparent_grayscale'],
                         $editable: $(this.odooEditor.editable), // Our parent is the root widget, we can't retrieve the editable section from it...
                         selectedColor: backgroundGradient || $(targetElement).css(eventName === "foreColor" ? 'color' : 'backgroundColor'),
+                        selectedTab: backgroundGradient ? 'gradients' : 'theme-colors',
                         withGradients: true,
                     });
                     colorpicker.on('custom_color_picked color_picked', null, ev => {
+                        if (hadSelection) {
+                            this.odooEditor.resetLostSelection();
+                        }
                         this._processAndApplyColor(eventName, ev.data.color);
-                        this.odooEditor.historyStep();
+                        colorpicker.setSelectedColor(null, ev.data.color, false);
                     });
                     colorpicker.on('color_hover color_leave', null, ev => {
-                        this._processAndApplyColor(eventName, ev.data.color);
+                        if (hadSelection) {
+                            this.odooEditor.resetLostSelection();
+                        }
+                        this.odooEditor.historyPauseSteps();
```

Comment:
This is in fact a fix unrelated to the branch right? => Dedicated commit. Also would allow us to validate it with the editor team, maybe a different PR would ease things for that. Thanks.

Reply:
Extracted to https://github.com/odoo/odoo/pull/76208